core/src/main/java/com/linecorp/armeria/client/endpoint/WeightRampingUpStrategyBuilder.java

```diff
@@ -0,0 +1,171 @@
+ * {@link Endpoint}s. The {@link Endpoint} is selected using weighted random distribution.
+ */
+public final class WeightRampingUpStrategyBuilder {
+
+    private static final long DEFAULT_RAMPING_UP_INTERVAL_MILLIS = 2000;
+    private static final int DEFAULT_NUMBER_OF_STEPS = 10;
+    private static final int DEFAULT_RAMPING_UP_ENTRY_WINDOW_MILLIS = 500;
+
+    private EndpointWeightTransition transition = EndpointWeightTransition.linear();
+
+    @Nullable
+    private ScheduledExecutorService executor;
+
+    private long rampingUpIntervalMillis = DEFAULT_RAMPING_UP_INTERVAL_MILLIS;
+    private int totalSteps = DEFAULT_NUMBER_OF_STEPS;
+    private long rampingUpTaskWindowMillis = DEFAULT_RAMPING_UP_ENTRY_WINDOW_MILLIS;
+
+    /**
+     * Sets computing the weight of an {@link Endpoint} during the transition with the specified
+     * {@link EndpointWeightTransition}. {@link EndpointWeightTransition#linear()} is used by default.
+     */
+    public WeightRampingUpStrategyBuilder transition(EndpointWeightTransition transition) {
+        this.transition = requireNonNull(transition, "transition");
+        return this;
+    }
+
+    /**
+     * Sets the specified {@link ScheduledExecutorService} that will be used to compute the weight of
+     * an {@link Endpoint} using {@link EndpointWeightTransition}. An {@link EventLoop} from
+     * {@link CommonPools#workerGroup()} is used by default.
+     */
+    public WeightRampingUpStrategyBuilder executor(ScheduledExecutorService executor) {
+        this.executor = requireNonNull(executor, "executor");
+        return this;
+    }
+
+    /**
+     * Sets the specified {@code rampingUpInterval}. The weight of an {@link Endpoint} is ramped up
+     * every {@code rampingUpInterval} until the number of ramping reaches to the
+     * {@link #totalSteps(int) totalSteps}.
+     * {@value DEFAULT_RAMPING_UP_INTERVAL_MILLIS} millis is used by default.
+     */
+    public WeightRampingUpStrategyBuilder rampingUpInterval(Duration rampingUpInterval) {
+        requireNonNull(rampingUpInterval, "rampingUpInterval");
+        return rampingUpIntervalMillis(rampingUpInterval.toMillis());
+    }
+
+    /**
+     * Sets the specified {@code rampingUpIntervalMillis}. The weight of an {@link Endpoint} is ramped up
+     * every {@code rampingUpIntervalMillis} until the number of ramping reaches to the
+     * {@link #totalSteps(int) totalSteps}.
+     * {@value DEFAULT_RAMPING_UP_INTERVAL_MILLIS} millis is used by default.
+     */
+    public WeightRampingUpStrategyBuilder rampingUpIntervalMillis(long rampingUpIntervalMillis) {
+        checkArgument(rampingUpIntervalMillis > 0,
+                      "rampingUpIntervalMillis: %s (expected: > 0)", rampingUpIntervalMillis);
+        this.rampingUpIntervalMillis = rampingUpIntervalMillis;
+        return this;
+    }
+
+    /**
+     * Sets the specified {@code totalSteps}. The weight of an {@link Endpoint} is ramped up
```

Comment:
Sets the number of steps to compute weights for a given endpoint while ramping up.